src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java

```diff
@@ -0,0 +1,91 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck;
+
+public class XpathRegressionUnnecessaryParenthesesTest extends AbstractXpathTestSupport {
```

Comment:
Could you add more tests please. From #5777:
> 2: Must have a UT for each log in the check. If the check has 3 separate log calls, then it must have 3 separate UTs.

Reply:
@shashwat70 
Still not done. There are 8 logs in the check but your test covers only 7. Looks like you missed this one:
```
if (assignDepth >= 1) {
    log(ast, MSG_ASSIGN);
}
```

Reply:
Could you please help me figure out what kind of errors are being logged by this log call? I figured something out but ended up finding a bug instead. #8239 
@rdiachenko

Reply:
@rdiachenko ping

Reply:
@shashwat70 , sorry for a delayed response.
Please take a look at this test - https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheckTest.java . It should cover that branch. Put a break point and try to debug yourself. If you still have problems please let me know.

Reply:
Done